core/dbt/parser/manifest.py

```diff
@@ -1482,6 +1656,74 @@ def process_node(config: RuntimeConfig, manifest: Manifest, node: ManifestNode):
     _process_docs_for_node(ctx, node)
 
 
+def write_publication_artifact(root_project: RuntimeConfig, manifest: Manifest):
+    # The manifest.json is written out in a task, so we're not writing it here
+
+    # build publication metadata
+    metadata = PublicationMetadata(
+        adapter_type=root_project.credentials.type,
+        quoting=root_project.quoting,
+    )
+
+    # get a list of public model ids first so it can be used in constructing dependencies
+    public_model_ids = []
+    for node in manifest.nodes.values():
+        if node.resource_type == NodeType.Model and node.access == AccessType.Public:
+            public_model_ids.append(node.unique_id)
+
+    set_of_public_model_ids = set(public_model_ids)
```

Comment:
refactoring suggestion: 
``` python
public_node_ids = {node.unique_id for node in manifest.nodes.values() if node.resource_type == NodeType.Model and node.access == AccessType.Public}
```

Reply:
Suggestion taken. Also changed the name of "public_model_ids" in the PublicationConfig to "public_node_ids", which appealed because it makes the difference a bit more clear.